proto/interchain_security/ccv/provider/v1/provider.proto

```diff
@@ -216,8 +216,9 @@ message InitTimeoutTimestamp {
 }
 
 message VscSendTimestamp {
-  uint64 vsc_id = 1;
-  google.protobuf.Timestamp timestamp = 2
+  string chain_id = 1;
```

Comment:
a new field ChainId is added, so that when exporting VscSendTimestamp to genesis, the chainID info is carried.

Reply:
Changing the order of fields inside a proto message is state breaking since this proto is used for storing state.

If you are adding a `chain_id` it should be done by adding it to the end (so it becomes `string chain_id = 3`).

I would advise against changing that structure, and instead change `GenesisState` of the provider like this:
```proto
type GenesisState struct {
	// strictly positive and set to 1 (DefaultValsetUpdateID) for a new chain
	ValsetUpdateId uint64 `protobuf:"varint,1,opt,name=valset_update_id,json=valsetUpdateId,proto3" json:"valset_update_id,omitempty"`
	// empty for a new chain
	ConsumerStates []ConsumerState `protobuf:"bytes,2,rep,name=consumer_states,json=consumerStates,proto3" json:"consumer_states" yaml:"consumer_states"`
	// empty for a new chain
	UnbondingOps []UnbondingOp `protobuf:"bytes,3,rep,name=unbonding_ops,json=unbondingOps,proto3" json:"unbonding_ops"`
	// empty for a new chain
	MatureUnbondingOps *types.MaturedUnbondingOps `protobuf:"bytes,4,opt,name=mature_unbonding_ops,json=matureUnbondingOps,proto3" json:"mature_unbonding_ops,omitempty"`
	// empty for a new chain
	ValsetUpdateIdToHeight []ValsetUpdateIdToHeight `protobuf:"bytes,5,rep,name=valset_update_id_to_height,json=valsetUpdateIdToHeight,proto3" json:"valset_update_id_to_height"`
	// empty for a new chain
	ConsumerAdditionProposals []ConsumerAdditionProposal `protobuf:"bytes,6,rep,name=consumer_addition_proposals,json=consumerAdditionProposals,proto3" json:"consumer_addition_proposals"`
	// empty for a new chain
	ConsumerRemovalProposals []ConsumerRemovalProposal `protobuf:"bytes,7,rep,name=consumer_removal_proposals,json=consumerRemovalProposals,proto3" json:"consumer_removal_proposals"`
	Params                   Params                    `protobuf:"bytes,8,opt,name=params,proto3" json:"params"`
	// empty for a new chain
	ValidatorConsumerPubkeys []ValidatorConsumerPubKey `protobuf:"bytes,9,rep,name=validator_consumer_pubkeys,json=validatorConsumerPubkeys,proto3" json:"validator_consumer_pubkeys"`
	// empty for a new chain
	ValidatorsByConsumerAddr []ValidatorByConsumerAddr `protobuf:"bytes,10,rep,name=validators_by_consumer_addr,json=validatorsByConsumerAddr,proto3" json:"validators_by_consumer_addr"`
	// empty for a new chain
	ConsumerAddrsToPrune []ConsumerAddrsToPrune `protobuf:"bytes,11,rep,name=consumer_addrs_to_prune,json=consumerAddrsToPrune,proto3" json:"consumer_addrs_to_prune"`
        InitTimeoutTImestamps ... // use the correct type
        VscSendTimestamps ... // use the correct type
}
```

That way we don't introduce state breaking changes, but have in mind that you would need to change the provider's Export genesis

```golang
// x/ccv/provider/keeper
func (k Keeper) ExportGenesis(ctx sdk.Context) *types.GenesisState {...}
```

Reply:
Hi, this way, we lost the chainID info for `InitTimeoutTImestamps` ?
how about create a new struct to wrap the  original `InitTimeoutTImestamps` with chainID for the genesis exporting.

Reply:
Yeah, unless it's necessary, try not to change the original structures as it will require state migration.

Reply:
We don't lose it, it's stored as part of the key in the state.

You can have a structure like this when exporting/importing:
```
type GenesisState struct {
        InitTimeoutTImestamps ... // use the correct type
        VscSendTimestamps []ExportedVscSendTimestamps
```

Where `ExportedVscSendTimestamps` looks like:
```golang
struct ExportedVscSendTimestamps {
    ChainId string
    Timestamps []VscSendTimestamps
}
```

Reply:
> Changing the order of fields inside a proto message is state breaking since this proto is used for storing state.

This is usually true, but in this case the proto unmarshal/marshal methods are not called, and the only place the `VscSendTimestamp` type is used is in `GetAllVscSendTimestamps`. Considering this, imo the best design is to simply add a chain ID field to the `VscSendTimestamp` type, and adjust the related keeper methods. 

This wouldn't be state breaking and would prevent us from adding another pb type just for exporting.

Reply:
but this is state breaking ? if we use a new struct to wrap `VscSendTimestamp` and add chainID, this is not state-breaking.

Reply:
yeah, the chainID is in store, we want to export the `VscSendTimestamp` plus `chainID` to genesis. this newly defined
```
struct ExportedVscSendTimestamps {
    ChainId string
    Timestamps []VscSendTimestamps
}

Reply:
`If i dump VscSendTimestamp to genesis, there is not chainID info in the genesis.`

Correct, I'm suggesting to add a chainID field to `VscSendTimestamp` instead of creating a new `ExportedVscSendTimestamps` type. This would not be state breaking. 

Overall if @mpoke and @MSalopek suggested the other way of doing this, that's fine with me and would still work

Reply:
adding chainID to `VscSendTimestamp` is api breaking ?

anway, can you review this PR ? Thank you!

Reply:
If possible, let's not reimplement this.

I'm suggesting to keep the changes already made. The benefit of the refactor is minimal.

Reply:
> adding chainID to VscSendTimestamp is api breaking ?

If API breaking means that the genesis JSON schema would change, then yes. That's not a problem tho

Re @MSalopek's comment, imo for tech debt reasons we should avoid adding shims/wrapper types unless they're absolutely needed to avoid a state migration. In this case there is no state migration to avoid.

I'll approve as to not slow everyone down, but since this is a `good first issue`, I just wanna make sure we understand the mechanics behind things being state breaking

Reply:
im just not sure about the impact of the api breaking change, so please let me know, i could change this PR if adding chainID to `VscSendTimestamp ` is better.

Reply:
People use "API breaking" in different ways, so I can't comment on impact of an api breaking change. But I can say it's ok to change the JSON schema for import/export genesis, and I believe any solution #612 would require we change that JSON schema

@MSalopek @mpoke lmk if I'm incorrect here

Re best solution for the PR, I'll leave that up to the other approver, as this thread has gotten long